lib/active_merchant/billing/gateways/commerce_hub.rb

```diff
@@ -109,6 +109,22 @@ def scrub(transcript)
 
       private
 
+      def add_three_d_secure(post, payment, options)
+        params = post[:additionalData3DS] = {}
```

Comment:
💬 Style/Comment:

Totally after you, but being more declarative than procedimental is always better:

```ruby
def add_three_d_secure(post, payment, options)
  return unless three_d_secure = options[:three_d_secure]

  post[:additionalData3DS] = {
    dsTransactionId: three_d_secure[:ds_transaction_id],
    authenticationStatus: three_d_secure[:authentication_response_status],
    serviceProviderTransactionId: three_d_secure[:three_ds_server_trans_id],
    acsTransactionId: three_d_secure[:acs_transaction_id],
    mpiData: {
      cavv: three_d_secure[:cavv],
      eci: three_d_secure[:eci],
      xid: three_d_secure[:xid]
    }.compact,
    version data: { recommendedVersion: three_d_secure[:version] }
  }.compact
end
```